app/views/unattended/provisioning_templates/snippet/kickstart_network_interface.erb

```diff
@@ -0,0 +1,107 @@
+<%#
+name: kickstart_network_interface
+model: ProvisioningTemplate
+snippet: true
+model: ProvisioningTemplate
+kind: snippet
+description: |
+  Generates network directive for a given interface. It is not expected to be used directly.
+-%>
+<%#
+  # Variables: iface, host, use_slaac, static, static6
+-%>
+<%if @iface.managed? -%>
+<%
+network_options = []
+nameservers = []
+subnet4 = @iface.subnet
+subnet6 = @iface.subnet6
+
+# device and hostname
+if @iface.bond? || @iface.bridge?
+  network_options.push("--device=#{@iface.identifier}")
+else
+  network_options.push("--device=#{@iface.mac || @iface.identifier}")
+end
+network_options.push("--hostname #{@host.name}")
+
+# single stack
+if subnet4 && !subnet6
+  network_options.push("--noipv6")
+elsif !subnet4 && subnet6
+  network_options.push("--noipv4")
+end
+
+# dual stack MTU check
+raise("IPv4 and IPv6 subnets have different MTU") if subnet4 && subnet6 && subnet4.mtu.present? && subnet6.mtu.present? && subnet4.mtu != subnet6.mtu
+
+# mtu method is taking into account both ipv4 and ipv6 stacks
+if @iface.respond_to?(:mtu) && @iface.mtu
+  network_options.push("--mtu=#{@iface.mtu}")
+end
+
+# IPv4
+if subnet4
+  if !subnet4.dhcp_boot_mode? || @static
+    network_options.push("--bootproto static")
+    network_options.push("--ip=#{@iface.ip}")
+    network_options.push("--netmask=#{subnet4.mask}")
+    network_options.push("--gateway=#{subnet4.gateway}")
+  elsif subnet4.dhcp_boot_mode?
+    network_options.push("--bootproto dhcp")
+  end
+
+  nameservers.concat(subnet4.dns_servers)
+end
+
+# IPv6
+if subnet6
+  if !subnet6.dhcp_boot_mode? || @static6
+    network_options.push("--ipv6=#{@iface.ip6}/#{subnet6.cidr}")
+    network_options.push("--ipv6gateway=#{subnet6.gateway}")
+  elsif subnet6.dhcp_boot_mode?
+    if @use_slaac
+      network_options.push("--ipv6 auto")
+    else
+      network_options.push("--ipv6 dhcp")
+    end
+  end
+
+  nameservers.concat(subnet6.dns_servers)
+end
+
+# bond
+if @iface.bond?
+  bond_slaves = @iface.attached_devices_identifiers.join(',')
+  network_options.push("--bondslaves=#{bond_slaves}")
+  network_options.push("--bondopts=mode=#{@iface.mode},#{@iface.bond_options.tr(' ', ',')}")
+end
+
+# bridge
+if @iface.bridge?
+  bridge_slaves = @iface.attached_devices_identifiers.join(',')
+  network_options.push("--bridgeslaves=#{bridge_slaves}")
+  # Currently no support for bridge options in the interface.
+end
+
+# VLAN (only on physical is recognized)
+if @iface.virtual? && @iface.tag.present? && @iface.attached_to.present?
```

Comment:
```
ksvalidator --version RHEL7   
The following problem occurred on line 18 of the kickstart file:

If network --interfacename starts with 'vlan', valid vlan id should follow.
```

Reply:
for EL8 & EL9 ksvalidator passed without issues

Reply:
Was the `iface.tag` empty for RHEL7 test by any chance?

Reply:
This is the ks code:
```
network --device=eth3 --hostname seven.example.com --bondslaves=eth0,eth1,eth2 --bondopts=mode=balance-rr,bopt0=00,bopt1=11 --nodns
network --device=00:aa:bb:cc:b7:10 --hostname seven.example.com --noipv6 --mtu=1500 --bootproto dhcp --vlanid=tagg --interfacename=vlantagg --nameserver=192.168.122.1
network --device=00:aa:bb:cc:53:ef --hostname seven.example.com --nodns
network --device=00:aa:bb:cc:4f:79 --hostname seven.example.com --nodns
network --device=00:aa:bb:cc:fc:8e --hostname seven.example.com --noipv6 --mtu=1500 --bootproto dhcp --nameserver=192.168.122.1
```
The tag value is `tagg`

Reply:
You should use a numeric value there: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/networking_guide/sec-naming_scheme_for_vlan_interfaces